internal/context/context.go

```diff
@@ -28,7 +30,7 @@ type ctxKey int
 const metadataCtxKey ctxKey = 0
 
 // New returns a context and cancel function for a given task message.
-func New(msg *base.TaskMessage, deadline time.Time) (context.Context, context.CancelFunc) {
+func New(base context.Context, msg *base.TaskMessage, deadline time.Time) (context.Context, context.CancelFunc) {
```

Comment:
Don't you need to replace the `context.Background()` call with this `base`?

Reply:
Ah! Missed that. Thanks for pointin it out